src/pytest_inmanta_lsm/load_generator.py

```diff
@@ -0,0 +1,218 @@
+        self.exception: typing.Optional[Exception] = None
+
+    def __enter__(self) -> None:
+        self.logger.debug("Creating new Thread")
+        self._thread = threading.Thread(target=self.between_callback, daemon=True, name="Thread-LG")
+        self.logger.debug("Starting %s", self._thread.name)
+        self._thread.start()
+
+    def __exit__(
+        self,
+        exc_type: typing.Optional[typing.Type[BaseException]],
+        exc: typing.Optional[BaseException],
+        traceback: typing.Optional[types.TracebackType],
+    ) -> None:
+        self.running = False
+        self.logger.debug("%s should stop", self.thread.name)
+        self.logger.debug("Stopping %s", self.thread.name)
+        self.thread.join(self.remote_orchestrator.client.timeout)
+        self.logger.debug("%s has been stopped", self.thread.name)
+
+        if self.exception is not None:
+            raise self.exception
+
+    @property
+    def thread(self) -> threading.Thread:
+        if self._thread is None:
+            raise RuntimeError("The thread is not initialized, this instance should be used in a context!")
+
+        return self._thread
+
+    def between_callback(self) -> None:
+        """
+        Will create the load in an async manner. This method will also save any exception that could have occurred in the
+        thread. This mechanism allows to raise an exception once the thread has joined, see the `__exit__` method.
+        """
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        try:
+            loop.run_until_complete(self.create_load())
+        except LoadException:
+            # We need to stop creating the load
+            self.logger.warning("%s - Load has been stopped!", self.thread.name)
+        except Exception as e:
+            self.exception = e
+        finally:
+            loop.close()
+
+    async def remote_call(self, call: functools.partial[typing.Coroutine[typing.Any, typing.Any, None]]) -> None:
+        """
+        First, it ensures that the Thread should still be running, otherwise it stops. Then, it interacts with the remote
+        orchestrator with the given callable function.
+
+        :param call: An asynchronous function that represents a specific request for the Remote Orchestrator
+        """
+        if not self.running:
+            raise LoadException()
+
+        try:
+            await call()
+        except Exception as e:
+            self.logger.warning("%s - encountered the following error:%s!", self.thread.name, str(e))
+        finally:
+            await asyncio.sleep(self.sleep_time)
+
+    async def create_load(self) -> None:
+        """
+        Create load until `running` flag is set to False
+        """
+        start_datetime = datetime.datetime.utcnow()
+        nb_datapoints = 15
+        # start_interval and end_interval should be at least <nb_datapoints> minutes separated from each other
```

Comment:
this is super confusing. It is about a specific input, not about the actual load generation. This should be made clear